com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -81,17 +85,45 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
         return StepPhase.POLL;
     }
 
+    private ApplicationResources getApplicationResources(ExecutionWrapper execution, CloudControllerClient client, CloudApplication app,

```

Comment:
This method explodes literally. 
Do you think whether there is a way of extracting it to a separate class in order to remove the complexity here?

Reply:
I would rather suggest to think in a way of either extracting it and also to split it into different methods responsible for different things. For instance, one for building the ApplicationResources, the other for reading and so on